### PR TITLE
Vertically center iOS text if lineHeight is set

### PR DESCRIPTION
**Pull request by tuckerconnelly**
First PR!!

This fixes https://github.com/facebook/react-native/issues/2991 :)

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @Kudo to be potential reviewers.

**Comment by charliesbox** [comment]
this is really helpful! please approve this pr

**Comment by mkonicek** [comment]
@tuckerconnelly Can you provide some screenshots from UIExplorer before and after? This might change how existing apps look so it is likely a breaking change though.

@nicklockwood, @javache What do you think?

File: Libraries/Text/RCTShadowText.m [hunk 1]
```diff
@@ -306,6 +306,11 @@ - (void)_setParagraphStyleOnAttributedString:(NSMutableAttributedString *)attrib
     CGFloat lineHeight = round(_lineHeight * fontSizeMultiplier);
     paragraphStyle.minimumLineHeight = lineHeight;
     paragraphStyle.maximumLineHeight = lineHeight;
+    // vertically center text
+    CGFloat fontSize = round(_fontSize * (_allowFontScaling && self.fontSizeMultiplier > 0.0 ? self.fontSizeMultiplier : 1.0));
+    [attributedString addAttribute:NSBaselineOffsetAttributeName
+                             value:[NSNumber numberWithFloat:lineHeight/2 - fontSize/2]
```
Comment (nicklockwood) [inline Libraries/Text/RCTShadowText.m]: Nit: can you replace

    [NSNumber numberWithFloat:lineHeight/2 - fontSize/2]

With

    @(lineHeight/2 - fontSize/2)

**Comment by nicklockwood** [comment]
@mkonicek I agree that this should be the default behaviour. Difficult to guess what the impact might be on existing apps though.

**Comment by facebook-github-bot-2** [comment]
@tuckerconnelly updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@tuckerconnelly updated the pull request.

**Comment by tuckerconnelly** [comment]
Boom, before:

screen shot 2016-05-29 at 4 39 24 pm

after:

screen shot 2016-05-29 at 4 40 35 pm

Also added code for adjusting the baseline when fontSize wasn't set.

**Comment by nihgwu** [comment]
any update? i'm looking forward this pr to be merged

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-0** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/277260495946830/int_phab) to review.

**Comment by bestander** [comment]
Shame for shipping this without looking at Travis build status https://travis-ci.org/facebook/react-native/jobs/133786669

This PR broke trunk

**Comment by bestander** [comment]
After talking to the teams internally, looks like we can't accept this change, have to revert it.
The problem is that it breaks the default behaviour and a lot of apps got broken.

**Comment by bestander** [comment]
Feel free to do another iteration and submit a new PR

**Comment by tuckerconnelly** [comment]
That's the nature of a breaking change lol

What would you want in another iteration?

**Comment by bestander** [comment]
Default lineheaight changes with this change, hence the snapshot test https://github.com/facebook/react-native/pull/8048/files

I guess it needs to be explicit and don't change default behaviour for now.

**Pull request closed without merging**